File: frontend/src/types/admin_cohort.ts
export interface StuckLearner {
  user_id: string;
  display_name: string;
  email_masked: string;
  last_activity_at: string | null;
  current_phase: number;
  submission_count: number;
  reason: 'no_submissions' | 'inactive_7d';
}

export interface TagHeatmapEntry {
  tag: string;
  average_score: number;
  submission_count: number;
}

export interface AdminCohortSummary {
  course_slug: string;
  course_title: string;
  enrolled_count: number;
  average_score: number | null;
  completion_rate: number;
  stuck_learners: StuckLearner[];
  tag_heatmap: TagHeatmapEntry[];
  cohort_label: string | null;
}

export interface AdminCohortLabels {
  items: string[];
}

File: backend/app/schemas/admin_cohort.py
"""Sprint 10 — admin cohort summary DTOs."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class StuckLearnerOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    user_id: uuid.UUID
    display_name: str
    email_masked: str
    last_activity_at: datetime | None
    current_phase: int
    submission_count: int
    reason: str = Field(description="no_submissions | inactive_7d")


class TagHeatmapEntryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    tag: str
    average_score: float
    submission_count: int


class AdminCohortSummaryOut(BaseModel):
    course_slug: str
    course_title: str
    enrolled_count: int
    average_score: float | None
    completion_rate: float
    stuck_learners: list[StuckLearnerOut]
    tag_heatmap: list[TagHeatmapEntryOut]
